Tests/UnitTests/Purchasing/OfferingsTests.swift

```diff
@@ -153,6 +153,55 @@ class OfferingsTests: TestCase {
         expect(offerings.current) == offerings["offering_a"]
     }
 
+    func testOfferingsWithMetadataIsCreated() throws {
+        @DefaultDecodable.EmptyDictionary
+        var metadata: [String: AnyDecodable] = [
+            "int": 5,
+            "double": 5.5,
+            "boolean": true,
+            "string": "five",
+            "array": ["five"],
+            "dictionary": [
+                "string": "five"
+            ]
+        ]
+
+        let annualProduct = MockSK1Product(mockProductIdentifier: "com.myproduct.annual")
+        let monthlyProduct = MockSK1Product(mockProductIdentifier: "com.myproduct.monthly")
+        let products = [
+            "com.myproduct.annual": StoreProduct(sk1Product: annualProduct),
+            "com.myproduct.monthly": StoreProduct(sk1Product: monthlyProduct)
+        ]
+        let offerings = try XCTUnwrap(
+            self.offeringsFactory.createOfferings(
+                from: products,
+                data: .init(
+                    currentOfferingId: "offering_a",
+                    offerings: [
+                        .init(identifier: "offering_a",
+                              description: "This is the base offering",
+                              packages: [
+                                .init(identifier: "$rc_six_month", platformProductIdentifier: "com.myproduct.annual")
+                              ],
+                             metadata: _metadata),
```

Comment:
Needing `@DefaultDecodable.EmptyDictionary var metadata: [String: AnyDecodable]` and then `_metadata` seems really weird to me but it was the only thing that made Xcode happy 🤷‍♂️ 

Here are screenshots of the errors I was getting otherwise 👇 

# 1. Without anything 

<img width="1026" alt="Screenshot 2023-05-23 at 5 57 56 AM" src="https://github.com/RevenueCat/purchases-ios/assets/401294/56f701aa-c97b-47fc-af05-ad804af2aead">

# 2. Without the underscore

<img width="1026" alt="Screenshot 2023-05-23 at 5 58 11 AM" src="https://github.com/RevenueCat/purchases-ios/assets/401294/e5067d7b-7c94-47ba-844c-6ce4c685c086">

Reply:
Oh. The right way to solve this is by creating a custom constructor for `OfferingsResponse.Offering`. But simpler is to change this line to `metadata: .init(wrappedValue: metadata)`, which is creating a `DefaultValue`.

Reply:
I know this is confusing, let me know if you need help with this.